### PR TITLE
Decompilation fix

### PR DESCRIPTION
**Pull request by mbriskar**
With this PR, every single .class file (A$1.class, A.class, A$InternalClass.class) will be connected with its decompiled part (A.java).

I also tried to create AbstractDecompiler so there is not that much duciplicits between the two implementations. However there is still some, so this is just a first step.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1178/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/550/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1192/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/564/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1193/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1193/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1193/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryModeTest/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryModeTest.testRunWindupTiny</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/565/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1194/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1194/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1194/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupMultiAppBasicTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupMultiAppBasicTest.testRunWindupMedium</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/566/

File: tests/src/test/java/org/jboss/windup/tests/application/WindupArchitectureTest.java [hunk 1]
```diff
@@ -197,6 +199,15 @@ JavaApplicationOverviewReportModel getApplicationOverviewReportByName(GraphConte
         return (JavaApplicationOverviewReportModel) reportModel;
     }
 
+    protected void allDecompiledFilesAreLinked(GraphContext context) {
```
Comment (mbriskar) [inline tests/src/test/java/org/jboss/windup/tests/application/WindupArchitectureTest.java]: this is the main test method testing this PR

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1198/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/570/

**Comment by jsight** [comment]
It looks good... thanks! Also, I think we have an intermittent test failure... I'm trying to figure it out now.

**Pull request merged**